applications/IgaApplication/custom_elements/iga_base_element.h

```diff
@@ -0,0 +1,166 @@
+/*
+//  KRATOS  _____________
+//         /  _/ ____/   |
+//         / // / __/ /| |
+//       _/ // /_/ / ___ |
+//      /___/\____/_/  |_| Application
+//
+//  Main authors:   Thomas Oberbichler
+*/
+
+#if !defined(KRATOS_IGA_BASE_ELEMENT_H_INCLUDED)
+#define KRATOS_IGA_BASE_ELEMENT_H_INCLUDED
+
+// System includes
+#include "includes/define.h"
+#include "includes/element.h"
+
+// External includes
+
+// Project includes
+
+namespace Kratos
+{
+
+template <std::size_t TDofsPerNode>
+class IgaBaseElement
+    : public Element
+{
+public:
+    using IgaBaseElementType = IgaBaseElement<TDofsPerNode>;
+
+    using Vector3D = BoundedVector<double, 3>;
+
+    using Element::Element;
```

Comment:
since this is the header-file of the baseclass it will propagate through also to the derived classes
=> it might be better to use typedef

Up to you, but I recently had to debug some stuff because of this and think its dangerous

Reply:
what would be the benefit of `typedef`?

Reply:
sorry... misunderstood the problem... the `using` is for inherit the constructors. AFAIK this can not be done with a `typedef`

Reply:
...and the propagate-effect is required